libglusterfs/src/glusterfs/list.h

```diff
@@ -28,7 +29,9 @@ list_add(struct list_head *new, struct list_head *head)
     new->next = head->next;
 
     new->prev->next = new;
-    new->next->prev = new;
+    if (new->next != NULL) {
```

Comment:
This shouldn't be needed. List pointers should never be NULL if correctly initialized. If a NULL is seen, the bug is the in caller, who has not correctly initialized the list, or has corrupted it.